### PR TITLE
Created types for json-schema-faker npm library

### PR DESCRIPTION
**Pull request by baremaximum**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [X] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## json-schema-faker ([<kbd>unpkg</kbd>](https://unpkg.com/browse/json-schema-faker@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'json-schema-faker'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. strToChars
1. tokenizeClass
1. error
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4712c2072a00f939a8d5a19aed6d48310675f58e
</p>

**Comment by JasonHK** [comment]
I suggest you fix your to-do list first.

- [x] Is `[x]`, not `[X ]` or `[ X]`.

File: types/json-schema-faker/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": ["es6"],
+        "noImplicitAny": true,
+        "strictNullChecks": true,
+        "esModuleInterop": true,
```
Comment (peterblazejewicz) [inline types/json-schema-faker/tsconfig.json:7]: You would probably have to remove that line, it's not enforced by linter, but anyway a rule, e.g.:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44595#discussion_r425400276
```suggestion
```
so effectively, imports would have to be rewritten I think to:
```ts
import jsf = require('json-schema-faker');
import { Schema } from 'json-schema-faker';
```

File: types/json-schema-faker/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
+// Type definitions for json-schema-faker 0.5
+// Project: https://github.com/json-schema-faker
```
Comment (peterblazejewicz) [inline types/json-schema-faker/index.d.ts:2]:
```suggestion
// Project: https://github.com/json-schema-faker/json-schema-faker
```
this is the organization's url, not the actual project url

File: types/json-schema-faker/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        | 'refDepthMin'
+        | 'refDepthMax'
+        | 'resolveJsonPath'
+        | 'reuseProperties'
+        | 'fillProperties'
+        | 'random'
+        | 'replaceEmptyByRandomValue';
+}
+
+declare function jsf(schema: jsf.Schema, refs?: string | jsf.Schema[]): any; // Deprecated.
```
Comment (peterblazejewicz) [inline types/json-schema-faker/index.d.ts:64]:
```suggestion
/** @deprecated calling JsonSchemaFaker() is deprecated, call either .generate() or .resolve()' */
declare function jsf(schema: jsf.Schema, refs?: string | jsf.Schema[]): any;
```

File: types/json-schema-faker/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+export = jsf;
```
Comment (peterblazejewicz) [inline types/json-schema-faker/index.d.ts:65]: AFAIK it looks to be an UMD module:

https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html

```suggestion
export as namespace jsf;
export = jsf;
```
(please first verify it locally with proper sample code/test)

**Review by peterblazejewicz — CHANGES_REQUESTED** [review]
@baremaximum It looks OK, just couple of comments, can you review them? Thanks!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@baremaximum thx for the contribution!

**Review by rbuckton — APPROVED** [review]

**Pull request merged**